backend/src/parser/TheParser.ts

```diff
@@ -86,15 +106,29 @@ export default class TheParser {
       decodeEntities: false,
     })
 
+    this.parseChildNodesStack.length = 0 // clear stack, should not be necessary, but just in case
     const parseResult = this.parseChildNodes(doc.childNodes)
     parseResult.mentions = [...new Set(parseResult.mentions)]
     return parseResult
   }
 
+  private isDisallowedTagNesting(child: string): boolean {
+    const disallowed = this.disallowedTagNesting[child]
+    if (!disallowed) {
+      return false
+    }
+    if (Array.isArray(disallowed)) {
+      for (const parent of disallowed) {
+        if (this.parseChildNodesStack.includes(parent)) return true
+      }
+      return false
+    }
+    return this.parseChildNodesStack.includes(disallowed)
+  }
```

Comment:
Maybe convert it to Sets and use [Set#intersection](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/intersection) ?

Reply:
I considered that, but there are two reasons why I chose this approach:
1. maintaining a set instead of the stack is not trivial, will required additional data structure with associated overhead
2. both arrays here are tiny (max depth of tag nesting in the typical content is rarely > 3), my intuition says the lower constant factor of the simple loops will win